quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -117,10 +130,45 @@ public String debug(final @PathParam("realmName") String realmName) throws IOExc
     @GET
     @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/" + PATH_FOR_TEST_CORS_IN_HEADERS)
     @Produces(MediaType.TEXT_PLAIN)
-    public Response test(final @PathParam("realmName") String realmName) {
-        Response.ResponseBuilder builder = Response.ok(PATH_FOR_TEST_CORS_IN_HEADERS + "-OK");
-        String origin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
-        builder.header(Cors.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+    public Response test(final @PathParam("realmName") String realmName, @DefaultValue("false") @QueryParam("frontEnd") boolean frontEnd) {
+        String text = "OK";
+        String corsOrigin = keycloakSession.getContext().getRequestHeaders().getHeaderString(Cors.ORIGIN_HEADER);
+        URI requestUri = keycloakSession.getContext().getUri().getRequestUri();
+        String requestOrigin = UriUtils.getOrigin(requestUri);
+        URI frontendUri = keycloakSession.getContext().getUri(UrlType.FRONTEND).getBaseUri();
+
+        if (frontEnd) {
+            boolean originMatches = requestOrigin.equals(UriUtils.getOrigin(frontendUri));
+            HttpHeaders requestHeaders = keycloakSession.getContext().getRequestHeaders();
+            boolean proxied = requestHeaders.getHeaderString("Forwarded") != null || requestHeaders.getHeaderString("X-Forwarded-For") != null;
+
+            if (!originMatches) { // might fail CORs checks
+                try {
+                    new URL(Configuration.getKcConfigValue(HostnameV2Options.HOSTNAME.getKey()).getValue());
+                    // a full url is ok
+                } catch (MalformedURLException e) {
+                    text = "Default origin check failing. Please check you proxy settings.";
+                    if (!keycloakSession.getContext().getHttpRequest().isProxyTrusted()) {
+                        text += " Note the proxy is not trusted.";
+                    }
+                    if (!proxied) {
+                        text += " No proxy headers are set on the request.";
+                    }
+                }
+            }
+
+            boolean https = requestUri.getScheme().equals("https");
+            if (https) {
+                // if reencrypt, then proxy headers may need set
+                // if passthrough, then proxy headers should not be set
+
+                // TODO: not sure if there is great way to do this. Would likely need to check that a connection to the frontend uses the same
+                // cert as what is coming in on this request
+            }
```

Comment:
This would rather be a security config issue, no? It's not really a hostname issue. Perhaps it doesn't belong to the hostname debug tool? It's a generic misconfig issue.

Reply:
> This would rather be a security config issue, no? 

Yes, but it relates to proxy-headers, so ideally it could be checked here as well.